pkg/apis/core/v1beta1/types.go

```diff
@@ -205,6 +207,24 @@ type DataVolumeSourceVDDK struct {
 	SecretRef string `json:"secretRef,omitempty"`
 }
 
+// DataVolumeSourceRef defines an indirect reference to the source of data for the DataVolume
+type DataVolumeSourceRef struct {
+	// The kind of the source reference, currently only DataSource is supported
+	Kind DataVolumeSourceRefKind `json:"kind"`
```

Comment:
Not sure we need/want type safety here.  `string` is what similar upstream types use.

Reply:
we are supporting only specific kinds (currently one...) so having just a `string` seems a bit odd.

Reply:
I think this is fine, but it doesn't prevent anyone from setting something else than DataSource.
So string would have the same effect, but having a type is just a bit more Go maybe.

Reply:
I don't see what type safety gets us here.  This value is 99% of the time going to be specified as a string in yaml.  Whether that gets deserialized as a string or some other type in Go doesn't really matter.

Take the dataSource field of PVCs for example.  kind is a string and can only have specific values

Reply:
Ok, this will be a `string`.